src/main/java/org/truffleruby/interop/RubyMessageResolution.java

```diff
@@ -236,6 +236,51 @@ protected Object access(VirtualFrame frame, DynamicObject object, Object name, O
 
     }
 
+    @Resolve(message = "REMOVE")
+    public static abstract class ForeignRemoveNode extends Node {
+
+        private final BranchProfile arrayReceiverProfile = BranchProfile.create();
+        private final BranchProfile hashReceiverProfile = BranchProfile.create();
+        private final BranchProfile errorProfile = BranchProfile.create();
+
+        @Child private CallDispatchHeadNode arrayDeleteAtNode = CallDispatchHeadNode.createOnSelf();
+        @Child private CallDispatchHeadNode hashDeleteNode = CallDispatchHeadNode.createOnSelf();
+        @Child private ForeignToRubyNode foreignToRubyNode = ForeignToRubyNode.create();
+
+        protected boolean access(VirtualFrame frame, DynamicObject object, Object name) {
+            if (RubyGuards.isRubyArray(object)) {
+                arrayReceiverProfile.enter();
+
+                try {
+                    arrayDeleteAtNode.call(frame, object, "delete_at", name);
+                } catch (RaiseException e) {
```

Comment:
It's not great to catch any Ruby exception and transform it in a UnknownIdentifierException not referring to the original cause.
Since the semantics are simple, how about just checking name is an Integer/Long here and otherwise let exceptions go through?

Reply:
I can see that point, but from an interop perspective, I see value in getting back an exception indicating "cannot remove with this name". And in that case, the end user doesn't need to know why. They just need to know that they can't remove that field/index/whatever from that object.

Reply:
I do agree the message should at least be logged in that case.

Reply:
> Since the semantics are simple, how about just checking name is an Integer/Long here and otherwise let exceptions go through?

That would achieve both goals: not hide potential bugs in delete_at and raise an exception when passing an invalid name (a non-integer here).

Reply:
I think we're disagreeing on the semantics. To me, the user tried to remove something with a particular name, it failed, they get back an exception they're expected handle indicating that name can't be used to remove something on that object.

Reply:
Looking at how Java interop works, I think we should throw `UnknownIdentifierException` if it isn't an integer, and let other exceptions pass through.